### ISSUE TITLE
`transaction::get_required_authorities(...)` got both owner authorities and active authorities of same account(s) when owner is required

### ISSUE THREAD

**Issue by vikramrajkumar** [bitshares/bitshares-core#197]
_From @abitmore on March 16, 2016 23:55_

When specified `owner` in `account_update_operation`, witness_node asks for both owner authorities and active authorities for signing the transaction. Worse case is when active authorities contain multiple keys but not all keys are in the wallet, even if the transaction is signed with owner key, sometimes it will still be rejected by witness_node. IMO active authorities is not needed in this case. 

We need a clear specification about owner/active authorities, not just concepts like https://bitshares.org/technology/dynamic-account-permissions/.

_Copied from original issue: cryptonomex/graphene#628_

**Comment by abitmore** [comment]
Need a test case to reproduce.

**Comment by abitmore** [comment]
Just figured out what this ticket was talking about.

TL;DR: not a serious bug, so won't fix.

The behavior:
1. when updating `owner`, if provided no signature, the transaction will be rejected with reason "Missing Active Authority"
2. when updating `owner`, if provided only signature of `active`, the transaction will be rejected with reason "Missing Owner Authority"
3. when updating `owner`, if provided both signatures of `owner` and `active`, the transaction will be accepted
4. when updating `owner`, if provided only signature of `owner`, the transaction will be accepted as well

So if ignored the last case, it looks like the operation need to be authorized by both owner and active authorities (mentioned in OP of this ticket). But it's not true, because the last case indicates that being authorized by owner is enough (the "worse case" description about this in OP is incorrect).

The reason (current implementation):
* by default, `fee_paying_account` is added to `required_active`, it's `account` in this op, so,
  * if not updating `owner`, `required_active` will be `[account]`, `required_owner` will be `[]`;
  * if updating `owner`, both `required_active` and `required_owner` will be `[account]`.
* In `transaction::verify_authority(...)`, `required_active` is checked before `required_owner`
  * when checking `required_active`, current `active` will be checked first, if not meet, current `owner` will be checked, so
    * the first case above, no signature, doesn't meet the requirement, so will throw an exception "Missing Active Authority"
    * other cases will pass this check
  * when checking `required_owner`, current `owner` will be checked, so
    * the second case above, only provided signature of active authorities, won't pass this check, will throw an exception "Missing Owner Authority"
    * the third case and the forth case will pass this check
* Signatures of public keys those have ever been checked will be accepted, so both the third and forth case will succeed, although in the third case the signature of active key is somehow unnecessary.

Ideally & strictly the transaction in the third case should be rejected for included unnecessary signature. However, since the behavior has been with us for 2 years, the cons (chain bloating) are not too big, and it's hard to change it without breaking things, so IMHO we can just leave it as is.

**Comment by abitmore** [comment]
We can still improve the `get_required_signatures` API, which should not break anything.

**Comment by abitmore** [comment]
Title changed.
Will fix the API(s) in PR #567.
Created a new ticket #580 for the signature verification issue.

**Comment by abitmore** [comment]
PR merged.
